core/src/main/java/org/elasticsearch/repositories/RepositoryData.java

```diff
@@ -44,9 +44,20 @@
  */
 public final class RepositoryData implements ToXContent {
 
-    public static final RepositoryData EMPTY = new RepositoryData(Collections.emptyList(), Collections.emptyMap());
+    /**
+     * The generation value indicating the repository has no index generational files.
+     */
+    public static final long EMPTY_REPO_GEN = -1L;
+    /**
+     * An instance initialized for an empty repository.
+     */
+    public static final RepositoryData EMPTY = new RepositoryData(EMPTY_REPO_GEN, Collections.emptyList(), Collections.emptyMap());
 
     /**
+     * The generational id of the index file from which the repository data was read.
+     */
+    private final long genId;
```

Comment:
I am not sure about this one. I feel like if it's part of `RepositoryData` it needs to be in json. Otherwise it cause things like `fromXContent` with additional parameters. Would it be a huge deal to pass gen id alone as an additional parameter instead of adding it to `RepositoryData` itself. I feel like it's not really a repositories concern, if you see what I mean.